docs/ux.md

```diff
@@ -0,0 +1,242 @@
+The name attribute references what the HPC administrator has defined.
+Mid-term, backend selection can be based on criteria other than a predefined name which refers to a spefific backend (e.g. by capacity and error rate qualifiers which help downselect between the defined set of backends).
+
+There might be additional environment variables required, depending on the backend type.
+
+#### Single QPU resources
+
+When a single QPU resource is used, SBATCH parameters will point to the QPU resource assigned to the application.
+Environment variables provided through the plugin can provide more details to the application (e.g. `SLURM_QPU_NAME` indicating which QPU resource is used eventually).
+
+```shell
+#SBATCH --time=100
+#SBATCH --output=<LOGS_PATH>
+#SBATCH --qpu=ibm_fez
+#SBATCH --... # other options
+
+srun ...
+```
+
+#### Multiple QPU resources
+
+Slurm qpu resources are given an identifier (in this example: *my_qpu_resource*) that can be referenced by applications e.g. to distinguish between several assigned resources.
+
+In this example, *my_qpu_resource* is the reference for usage of the quantum resource in the application.
+If the resource changes (e.g. from *ibm_fez* to *ibm_marrakesh*), the application code does not have to change -- but the SBATCH parameters will have to change.
+
+```shell
+#SBATCH --time=100
+#SBATCH --output=<LOGS_PATH>
+#SBATCH --qpu=my_qpu_resource1
+#SBATCH --qpu-name=my_local_qpu
+#SBATCH --qpu=my_qpu_resource2
+#SBATCH --qpu-name=ibm_fez
+#SBATCH --qpu=my_qpu_resource3
+#SBATCH --qpu-name=ibm_marrakesh
+#SBATCH --... # other options
+
+srun ...
+```
+
+### HPC application scope
+
+HPC applications refer to the slurm QPU resources assigned to the slurm job.
+
+(Please check whether this flow clicks. We want to reference the slurm resource name, in case several resources are defined; also we ideally avoid specifying whether it's a DirectAccess or QiskitRuntimeService (Pasqal code is probably not very portable to other types at this time); also it seems we wanted to avoid a backend object and go with target instead)
+
+#### Single QPU resources
+
+For single QPU scenarios, the QPU will be identified and used automatically.
+
+```python
+from qrmi import IBMDirectAccessSamplerV2 as SamplerV2
+
+# Generate transpiler target from backend configuration & properties
+target = get_target()
+pm = generate_preset_pass_manager(
+    optimization_level=1,
+    target=target,
+)
+
+
+sampler = SamplerV2(target=target)
+```
+
+#### Multiple QPU resources
+
+A named identifier represents the QPU resource and can be used in the source code.
+
+```python
+from qrmi import IBMDirectAccessSamplerV2
+from qrmi import IBMQiskitRuntimeServiceSamplerV2
+
+# Generate transpiler target from backend configuration & properties
+target1 = get_target(slurm_resource_name=“my_qpu_resource1”)
+target2 = get_target(slurm_resource_name=“my_qpu_resource2”)
+target2 = get_target(slurm_resource_name=“my_qpu_resource3”)
```

Comment:
My suggestion above would result in this being the following:
```python
num_targets = os.environ["SLURM_QPU_COUNT"]
targets = [get_target(id=i) for i in range(num_targets)]
```
where
```python
def get_target(index: int = 0) -> Target:
    target_names = os.environ["SLURM_QPU_NAMES"].split(";")  # assuming we have a ";"-separated list of names
    target_name = target_names[index]
    # obtain all target properties and whatever we return here
    return Target.from_name(target_name)
```

Reply:
If we do not like the `SLURM_QPU_NAMES` we could also have separate env variables for each name be defined:
```
SLURM_QPU_NAME0 = #str
SLURM_QPU_NAME1 = #str
...  # continues up to `SLURM_QPU_NAME{SLURM_QPU_COUNT}`
```
In this case we would need a `SLURM_QPU_COUNT` variable which determines the number of variables defined.